app/Helpers/database/achievement.php

```diff
@@ -416,18 +399,10 @@ function updateAchievementFlag(int|string|array $achID, int $newFlag): void
         return;
     }
 
-    $achievements->update(['Flags' => $newFlag]);
-
     $updatedAchievements = Achievement::whereIn('ID', $achievementIDs)->get();
-
     foreach ($updatedAchievements as $achievement) {
-        if ($newFlag === AchievementFlag::OfficialCore) {
-            AchievementPublished::dispatch($achievement);
-        }
-
-        if ($newFlag === AchievementFlag::Unofficial) {
-            AchievementUnpublished::dispatch($achievement);
-        }
+        $achievement->Flags = $newFlag;
+        $achievement->save();
```

Comment:
Saving model individually will trigger model events. Batch updates like `$achievements->update(['Flags' => $newFlag]);` previously do not trigger model events.